### PR TITLE
fix(es/minifier): Don't remove used var decl

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/7194.

---

# Invesitgation:

The problematic code is

https://github.com/swc-project/swc/blob/610e1bb581a04fe9b871dd7afa0cf81ab9ee7fa8/crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs#L1217-L1218

### CODE REVIEW

**Comment by kdy1** [comment]
@Austaras Can you take a look?

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

**Comment by Austaras** [comment]
I believe a more fine grained solution would be most optimal -- the pure mark only marks the function call, but not the call arguments, but here we ignore usage in arguments as well.

File: crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs [hunk 1]
```diff
@@ -1214,13 +1214,35 @@ where
...
-                    let can_ignore =
-                        matches!(&**init, Expr::Call(call) if call.span.has_mark(marks.pure));
+
                     let id = id.to_id();
-                    self.used_recursively
-                        .insert(id.clone(), RecursiveUsage::Var { can_ignore });
-                    e.init.visit_with(&mut *self.with_ctx(ctx));
-                    self.used_recursively.remove(&id);
+
+                    match init {
```
Comment (kdy1) [inline crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs:1220]: @Austaras Did you mean something like this?

Comment (Austaras) [reply crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs]: I can't see changes, seems like they're force pushed.

**Review by jridgewell — APPROVED** [review]

**Pull request merged**